### ISSUE TITLE
cannot request paths containing escaped slash, i.e. '/api/exchanges/%2f/amq.direct'

### ISSUE THREAD

**Issue by kpe** [Kong/unirest-java#160]
Trying to access the RabbitMO management API, I figured out, that it is impossible to send a request to an URL containing an escaped slash '/', i.e. '/api/exchanges/%2f/amq.direct'
That's mainly because the java URI(scheme,...,...) constructor would do a toString() first qouting/escaping all parameters and than parse the resulting string; during the toString() however the path is quoted/encoded, so a a string like '%2f' is written out as '%252f'. 
And specifying the path as '/api///whatever' is not a solution either, as the '/' is valid path character, and does not get escaped.

One might surely argue, that this is Java URI problem, however it prevents unirest-java to be used with URIs containing an escaped '/'.

**Comment by stangls** [comment]
Inside `com.mashape.unirest.http.HttpClientHelper` method `prepareRequest`, the URL is split up into an URI where the path is url-decoded. This seems not correct.

**Comment by stangls** [comment]
@kpe can you please try if this dependency fixes the problem?
https://jitpack.io/#stangls/unirest-java/e92f6db882

**Comment by EricWittmann** [comment]
Any reason why https://github.com/Mashape/unirest-java/pull/203 wasn't merged?  This still seems to be a problem.  We just ran into the same issue trying to use the GitLab API (some path items are URL encoded, but the HttpClientHelper decodes them).  This is still the problematic line of code:

https://github.com/Mashape/unirest-java/pull/203

I don't see a way around it other than using Apache HTTP Client directly...

**Comment by stevespringett** [comment]
I also ran into this problem recently, specifically when using the npm api with scoped packages in the URL. Since this project is dead, I checked out the [OpenUnirest](https://github.com/OpenUnirest/unirest-java) fork, which is very actively maintained, and that project has fixed this issue. Best of all, there were no code changes necessary, only changing the imports since the package names have changed. 

So simply migrating to OpenUnirest has solved this issue for me.
